### PR TITLE
Add a waitlist page for phone masking

### PR DESCRIPTION
**Pull request by Vinnl**
# New feature description

I basically re-used the existing waitlist page, which I extracted into a separate component (see the first commit) - the only difference is a couple of strings, and the ID for the newsletter in Basket.

I still have to get final signoff on the strings and all that, and I might need to add another waitlist for the bundle, but I can do that in a separate PR if I get clarity on that after code review.

# Screenshot (if applicable)

image

# How to test

[Visit `/phone/waitlist/`](https://deploy-preview-2939--fx-relay-demo.netlify.app/phone/waitlist/).

# Checklist (Definition of Done)
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [ ] All acceptance criteria are met. (Bundle waitlist isn't there - not sure if needed.)
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] I've added or updated relevant docs in the docs/ directory
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.
- [x] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [ ] l10n changes have been submitted to the l10n repository, if any.

### CODE REVIEW

File: frontend/src/components/waitlist/WaitlistPage.tsx [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+  const onSubmit: FormEventHandler = async (event) => {
+    event.preventDefault();
+
+    if (!email) {
+      // Modern browsers should prevent this from happening, due to the `required` attribute:
+      console.error(
+        "WaitlistPage.onSubmit: form submitted while required input field was empty."
+      );
+      return;
+    }
```
Comment (lloan) [inline frontend/src/components/waitlist/WaitlistPage.tsx:59]: question (non blocking): If for any reason the user is able to submit the form, should the user be told as such, visually? Is this what we would use our toast notifications for?

**Review by lloan — APPROVED** [review]
Looks good, just left one question. We're just waiting on strings for this right?

File: frontend/src/components/waitlist/WaitlistPage.tsx (hunk 1 above)
Comment (Vinnl) [reply frontend/src/components/waitlist/WaitlistPage.tsx:59]: That's a good question. That would indeed be best, and while whether using a toast is a UI design question (but my assumption would be yes), we could definitely do something like that. That said, given [browser support for `required`](https://caniuse.com/mdn-api_htmlinputelement_required), I don't think there's that big a risk of the user being able to submit the form anyway.

And yes, I think final strings came in while I was on holiday, so I'll add those to the PR.

Comment (Vinnl) [reply frontend/src/components/waitlist/WaitlistPage.tsx:59]: OK, copy-wise the only change was casing. There's also copy for a bundle waitlist page, but I'll submit a separate PR that adds that page.

**Pull request merged**